core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileSystem.java

```diff
@@ -34,9 +37,14 @@ public LocalCacheFileSystem(FileSystem fs, FileSystemContext fsContext) {
   }
 
   @Override
-  public FileInStream openFile(AlluxioURI path, OpenFilePOptions options) {
+  public FileInStream openFile(AlluxioURI path, OpenFilePOptions options)
+      throws IOException, AlluxioException {
     // TODO(calvin): We should add another API to reduce the cost of openFile
+    CacheManager cacheManager = mFsContext.getCacheManager();
+    if (cacheManager == null) {
```

Comment:
We should log warn the fallback. However, if you are concerned that the log is too frequent, we can check if `getCacheManager()` equals `null` at the factory method `alluxio.client.file.FileSystem.Factory#create`; in case yes, fallback to returning a normal filesystem instance even `USER_LOCAL_CACHE_ENABLED` is true and `LOG.warn` the fallback?
Then in here, we can simply `Precondition.checkNotNull(cacheManager)`.

Reply:
In `FileSystemContext::getCacheManager`, we already log a warning when the cache manager fails to initialize. Would that be sufficient?